src/app/clusters/resource-monitoring-server/resource-monitoring-server.h

```diff
@@ -0,0 +1,161 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+#include <app/clusters/resource-monitoring-server/resource-monitoring-cluster-objects.h>
+#include <app/util/af.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ResourceMonitoring {
+
+class Instance : public CommandHandlerInterface, public AttributeAccessInterface
+{
+
+public:
+    void LoadPersistentAttributes();
+
+    /**
+     * Initialise the Resource Monitoring cluster.
+     *
+     * @die                                     If the cluster ID given is not a valid Resource Monitoring cluster ID.
+     * @die                                     If the endpoint and cluster ID have not been enabled in zap.
+     * @return CHIP_ERROR_INVALID_ARGUMENT      If the CommandHandler or Attribute Handler could not be registered.
+     * @return CHIP_ERROR_INCORRECT_STATE       If the CommandHandler was already registered
+     * @return CHIP_ERROR_INCORRECT_STATE       If the registerAttributeAccessOverride fails.
+     * @return CHIP_ERROR                       If the AppInit() method returned an error. This is application specific.
+     *
+     * @return CHIP_NO_ERROR                    If the cluster was initialised successfully.
+     */
+    CHIP_ERROR Init();
+
+    // CommandHandlerInterface
+    void InvokeCommand(HandlerContext & ctx) override;
+    CHIP_ERROR EnumerateAcceptedCommands(const ConcreteClusterPath & cluster, CommandIdCallback callback, void * context) override;
```

Comment:
Just move these to private?  They're not really "public" API, right?  In terms of the API that we expect the subclasses to actually use.

Reply:
Made them private